tests/meson.build

```diff
@@ -0,0 +1,347 @@
+tests_exe = executable(
+  'msft_proxy_tests',
+  files(
+    'proxy_creation_tests.cpp',
+    'proxy_dispatch_tests.cpp',
+    'proxy_fmt_format_tests.cpp',
+    'proxy_format_tests.cpp',
+    'proxy_integration_tests.cpp',
+    'proxy_invocation_tests.cpp',
+    'proxy_lifetime_tests.cpp',
+    'proxy_reflection_tests.cpp',
+    'proxy_regression_tests.cpp',
+    'proxy_rtti_tests.cpp',
+    'proxy_traits_tests.cpp',
+    'proxy_view_tests.cpp',
+  ),
+  implicit_include_directories: false,
+  dependencies: [msft_proxy4_dep, gtest_dep, fmt_dep],
+  build_by_default: false,
+)
+
+autogen_tests = custom_target(
+  command: [
+    autogen_tests_exe,
+    custom_target(
+      command: [tests_exe, '--gtest_list_tests', '--gtest_output=json:@OUTPUT@'],
+      output: 'tests.json',
+    ),
+    meson.current_source_dir() / 'meson.build',
+  ],
+  output: 'autogen.stamp',
+  build_by_default: true,
+)
+
+#pragma autogen push
+tests = {
```

Comment:
What about something like:

```
test(
  'ProxyTests',
  tests_exe,
  protocol: 'gtest',
  suite: 'Proxy',
)
```

... so that the list can be eliminated.